tests/test_processor.py

```
@@ -27,3 +41,72 @@ def test_is_sidecar_file_negative() -> None:
+    image_path.write_bytes(b'\xff\xd8\xff\xe0')  # JPEG magic bytes
+    
+    # Create corresponding JSON file
+    json_path = tmp_path / "photo.png.supplemental-metadata.json"
+    json_data = {"title": "photo.png"}
+    json_path.write_text(json.dumps(json_data), encoding='utf-8')
+    
+    # Mock Path.unlink to raise an error (simulating read-only file)
+    original_unlink = Path.unlink
+    def mock_unlink(self):
+        if self.name.endswith('.supplemental-metadata.json') and 'photo.png' in str(self):
+            raise OSError("Permission denied")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)